### ISSUE TITLE
Zoning keeps looping over the device

### ISSUE THREAD

**Issue by iustin** [axboe/fio#738]
I think I still see the issue mentioned in https://www.spinics.net/lists/fio/msg03939.html, but in my case with sequential I/O. While the code in setup_strided_zone_mode is slightly different, it still has the same "wrap from the beginning if we exceed the file size" behaviour.

I'm using fio-3.12 and this issue prevents me from doing a simple "give me a zoned bandwidth log for a device", due to the looping which results in infinite runtime.

My job file:
```
[global]
bs=512k
direct=1
rw=read
ioengine=libaio
iodepth=2
zonemode=strided
zonerange=1g
zonesize=1g
zoneskip=1000g
write_bw_log=bw-log

[zoning]
name=zoning-log
filename=${DRIVE}
```

Would be very glad to hear this is just misconfiguration on my part…

**Comment by sitsofe** [comment]
@bvanassche any ideas on this one?

**Comment by bvanassche** [comment]
@damien-lemoal or @dmitry-fomichev, can you have a look?

**Comment by dmitry-fomichev** [comment]
will take a look...

**Comment by sitsofe** [comment]
@dmitry-fomichev any luck?

**Comment by dmitry-fomichev** [comment]
> 
> 
> @dmitry-fomichev any luck?

Looking now... will update in a few days

**Comment by dmitry-fomichev** [comment]
Ran the workload file from the description under the latest fio code and the test gets properly terminated. Here is the log result -
```
[root@xxxxx fio]# fio 738.fio
zoning-log: (g=0): rw=read, bs=(R) 512KiB-512KiB, (W) 512KiB-512KiB, (T) 512KiB-512KiB, ioengine=libaio, iodepth=2

fio-3.16-23-g2e97f
Starting 1 process
Jobs: 1 (f=1): [R(1)][100.0%][r=265MiB/s][r=529 IOPS][eta 00m:00s]
zoning-log: (groupid=0, jobs=1): err= 0: pid=25537: Thu Oct 17 13:51:20 2019
  read: IOPS=526, BW=263MiB/s (276MB/s)(279GiB/1087088msec)
    slat (usec): min=23, max=571, avg=76.92, stdev= 5.56
    clat (usec): min=1988, max=15067, avg=3713.54, stdev=313.26
     lat (usec): min=2065, max=15156, avg=3791.36, stdev=313.18
    clat percentiles (usec):
     |  1.00th=[ 3523],  5.00th=[ 3556], 10.00th=[ 3589], 20.00th=[ 3589],
     | 30.00th=[ 3589], 40.00th=[ 3589], 50.00th=[ 3589], 60.00th=[ 3589],
     | 70.00th=[ 3621], 80.00th=[ 3654], 90.00th=[ 4146], 95.00th=[ 4424],
     | 99.00th=[ 4948], 99.50th=[ 5080], 99.90th=[ 5473], 99.95th=[ 5669],
     | 99.99th=[ 8586]
   bw (  KiB/s): min=34796, max=263621, per=52.67%, avg=141984.65, stdev=9650.54, samples=572337
   iops        : min=  314, max=  544, avg=526.43, stdev= 9.35, samples=2174
  lat (msec)   : 2=0.01%, 4=86.09%, 10=13.91%, 20=0.01%
  cpu          : usr=0.88%, sys=4.21%, ctx=572903, majf=0, minf=4759
  IO depths    : 1=0.1%, 2=100.0%, 4=0.0%, 8=0.0%, 16=0.0%, 32=0.0%, >=64=0.0%
     submit    : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     complete  : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     issued rwts: total=572337,0,0,0 short=0,0,0,0 dropped=0,0,0,0
     latency   : target=0, window=0, percentile=100.00%, depth=2

Run status group 0 (all jobs):
   READ: bw=263MiB/s (276MB/s), 263MiB/s-263MiB/s (276MB/s-276MB/s), io=279GiB (300GB), run=1087088-1087088msec

Disk stats (read/write):
  sda: ios=572337/0, merge=0/0, ticks=2128292/0, in_queue=1802450, util=57.07%
[root@xxxxx fio]#
```
After looking at the code referred in the link in the issue description, i.e. 4c8be5b ("Fix bug with zone and zone skipping and io_limit"), it is clear that this issue has been fixed by the following commit: 04bc85a. Therefore, this one can be closed as resolved.

**Comment by iustin** [comment]
Yes, tha looks likely. Note I tested with fio 3.12, but the quoted fix was included in 3.16 only, hence me stil hitting this issue early this year. Thanks!
